File: src/plugin/combat/special/melee/AbyssalWhipSpecialAttackPlugin.java
package plugin.combat.special.melee;

import org.redrune.game.content.combat.player.AbstractCombatStyle;
import org.redrune.game.entity.actor.Actor;
import org.redrune.game.entity.actor.mask.Animation;
import org.redrune.game.entity.actor.mask.Graphics;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.plugin.combat.SpecialAttackPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 9/5/2017
 */
public class AbyssalWhipSpecialAttackPlugin extends SpecialAttackPlugin {
	
	private static final Animation ANIMATION = new Animation(11971);
	
	private static final Graphics GRAPHICS = new Graphics(2108, 0, 96);
	
	@Override
	public int[] getWeaponIds() {
		return arguments(4151, 15442, 15443, 15444, 15441, 23691);
	}
	
	@Override
	public void fire(Player source, Actor target, AbstractCombatStyle style) {
		source.setNextAnimation(ANIMATION);
		target.setNextGraphics(GRAPHICS);
		if (target instanceof Player) {
			Player p2 = (Player) target;
			p2.setRunEnergy(p2.getRunEnergy() > 25 ? p2.getRunEnergy() - 25 : 0);
		}
		style.sendHit(source, target, style.getCalculator().getMaximumHit(source,1), style.getRandomDamage(source, target, 1), 0);
	}
}

File: src/main/java/plugin/interaction/combat/special/melee/AbyssalWhipSpecialAttackPlugin.kt
package plugin.interaction.combat.special.melee

import org.redrune.game.content.entity.actor.combat.player.AbstractCombatStyle
import org.redrune.game.content.plugin.combat.SpecialAttackPlugin
import org.redrune.game.entity.actor.Actor
import org.redrune.game.entity.actor.mask.Animation
import org.redrune.game.entity.actor.mask.Graphics
import org.redrune.game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 9/5/2017
 */
class AbyssalWhipSpecialAttackPlugin : SpecialAttackPlugin() {
    override fun getWeaponIds(): IntArray {
        return arguments(4151, 15442, 15443, 15444, 15441, 23691)
    }

    override fun fire(source: Player, target: Actor, style: AbstractCombatStyle) {
        source.nextAnimation = ANIMATION
        target.setNextGraphics(GRAPHICS)
        if (target is Player) {
            val p2 = target
            p2.attributes.setRunEnergy(if (p2.attributes.runEnergy > 25) p2.attributes.runEnergy - 25 else 0)
        }
        style.sendHit(
            source,
            target,
            style.calculator.getMaximumHit(source, 1.0),
            style.getRandomDamage(source, target, 1.0),
            0
        )
    }

    companion object {
        private val ANIMATION = Animation(11971)
        private val GRAPHICS = Graphics(2108, 0, 96)
    }
}